libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -559,12 +602,95 @@ AP_GPS_SBF::process_message(void)
                             (unsigned int)(RxError & RX_ERROR_MASK), (unsigned int)(temp.RxError & RX_ERROR_MASK));
         }
         RxError = temp.RxError;
+        if ((ExtError & EXT_ERROR_MASK) != (temp.ExtError & EXT_ERROR_MASK)) {
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO, "GPS %u: SBF external error changed (0x%08x/0x%08x)", (unsigned int)(state.instance + 1),
+                            (unsigned int)(ExtError & EXT_ERROR_MASK), (unsigned int)(temp.ExtError & EXT_ERROR_MASK));
+        }
+        ExtError = temp.ExtError;
+#if AP_MAVLINK_MSG_GNSS_INTEGRITY_ENABLED
+        state.system_errors = static_cast<uint32_t>(AP_GPS::Errors::GPS_SYSTEM_ERROR_NONE);
+
+        if (ExtError & DIFFCORRERROR) {
+            state.system_errors |= static_cast<uint32_t>(AP_GPS::Errors::INCOMING_CORRECTIONS);
+        }
+
+        for (const auto &map_entry : sbf_error_map) {
+            if (RxError & map_entry.sbf_error_bit) {
+                state.system_errors |= static_cast<uint32_t>(map_entry.ap_error_enum);
+            }
+        }
+#endif  // AP_MAVLINK_MSG_GNSS_INTEGRITY_ENABLED
+        break;
+    }
+
+    case RFStatus:
+    {
+        const msg4092 &temp = sbf_msg.data.msg4092u;
+        check_new_itow(temp.TOW, sbf_msg.length);
+#if AP_MAVLINK_MSG_GNSS_INTEGRITY_ENABLED
+        if (temp.Flags==0) {
+            state.spoofing_state = static_cast<uint8_t>(AP_GPS::Spoofing::OK);
+        }
+        else {
+            state.spoofing_state = static_cast<uint8_t>(AP_GPS::Spoofing::DETECTED);
+        }
+        state.jamming_state = static_cast<uint8_t>(AP_GPS::Jamming::OK);
+
+        if (temp.SBLength < sizeof(RFStatusRFBandSubBlock)) {
+            break;
+        }
+
+        const uint8_t *p = (const uint8_t *)&temp.RFBand;
+        const uint8_t *packet_end = (const uint8_t *)&temp + (sbf_msg.length - 8);
+
+        for (uint8_t i = 0; i < temp.N; i++) {
+            if (p + temp.SBLength > packet_end) {
+                break;
+            }
+
+            const RFStatusRFBandSubBlock &rf_band_data = *(const RFStatusRFBandSubBlock *)p;
+
+            switch (rf_band_data.Info & (uint8_t)0b1111) {
+            case 1:
+            case 2:
+                // As long as there is indicated but unmitigated spoofing in one band, don't report the overall state as mitigated
+                if (state.jamming_state == static_cast<uint8_t>(AP_GPS::Jamming::OK)) {
+                    state.jamming_state = static_cast<uint8_t>(AP_GPS::Jamming::MITIGATED);
+                }
+                break;
+            case 8:
+                state.jamming_state = static_cast<uint8_t>(AP_GPS::Jamming::DETECTED);
+                break;
+            }
+            p += temp.SBLength;
+        }
+#endif  // AP_MAVLINK_MSG_GNSS_INTEGRITY_ENABLED
         break;
     }
+
+    case GALAuthStatus:
+    {
+        const msg4245 &temp = sbf_msg.data.msg4245u;
+        check_new_itow(temp.TOW, sbf_msg.length);
+#if AP_MAVLINK_MSG_GNSS_INTEGRITY_ENABLED
```

Comment:
Similar to the above - should we know about GALAuthStatus at all if we're not doing integrity stuff?